[25.0] Directory converters: fix for flat zip files

bernt-matthias opened the pull request:
Was wondering converters should be added automatically for subclassed datatypes?

## How to test the changes?
(Select all options that apply)
- [x] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [ ] This is a refactoring of components with existing test coverage.
- [ ] Instructions for manual testing are as follows:
  1. [add testing steps and prerequisites here if you didn't write automated tests covering all your changes]

## License
- [x] I agree to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT).

bernt-matthias committed:
bump requirement of directory converters

with conda I get

```
Traceback (most recent call last):
  File "<string>", line 1, in <module>
    from galaxy.util.compression_utils import CompressedFile; CompressedFile('/tmp/tmp3v2nuot4/files/a/9/7/dataset_a97aebb4-46f7-47a9-b418-11ff1deeed74.dat').extract('.')
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/berntm/miniforge3/envs/__galaxy-util@23.2.1/lib/python3.13/site-packages/galaxy/util/compression_utils.py", line 29, in <module>
    from .checkers import (
    ...<2 lines>...
    )
  File "/home/berntm/miniforge3/envs/__galaxy-util@23.2.1/lib/python3.13/site-packages/galaxy/util/checkers.py", line 20, in <module>
    from galaxy.util.image_util import image_type
  File "/home/berntm/miniforge3/envs/__galaxy-util@23.2.1/lib/python3.13/site-packages/galaxy/util/image_util.py", line 3, in <module>
    import imghdr
ModuleNotFoundError: No module named 'imghdr'
```

bernt-matthias commented on lib/galaxy/datatypes/converters/archive_to_directory.xml:
```diff
@@ -22,6 +22,7 @@
             <option value="directory">directory</option>
             <option value="zarr">zarr</option>
             <option value="ome_zarr">ome_zarr</option>
+            <option value="lexicmap_index">lexicmap_index</option>
```
Wondering why the tar2directory converter does not have this parameter. Should I add it?

Edit: Or should the tar2directory converter be replaced by this tool?
EditEdit: Ahh it already is.

mvdbeek commented on lib/galaxy/config/sample/datatypes_conf.xml.sample:
```diff
@@ -329,6 +329,7 @@
     <datatype extension='qiime2.tabular' type="galaxy.datatypes.qiime2:QIIME2Metadata" display_in_upload="true"/>
     <datatype extension="zip" type="galaxy.datatypes.binary:CompressedZipArchive" display_in_upload="true">
         <converter file="archive_to_directory.xml" target_datatype="directory" />
+        <converter file="archive_to_directory.xml" target_datatype="lexicmap_index"/>
```
This doesn't really make sense to me, that means every zip file is also a lexicmap_index.

mvdbeek commented on lib/galaxy/datatypes/converters/tar_to_directory.xml:
```diff
@@ -1,7 +1,7 @@
     <!-- Don't use tar directly so we can verify safety of results - tar -xzf '$input1'; -->
     <requirements>
-        <requirement type="package" version="23.2.1">galaxy-util</requirement>
+        <requirement type="package" version="25.0.2">galaxy-util</requirement>
```
Is this the fix you're referring to in https://github.com/galaxyproject/tools-iuc/pull/7257/files#r2323290063 ?

bernt-matthias replied:
The fix is in the command block. 

The bump is essentially a backport of https://github.com/galaxyproject/galaxy/pull/20834

bernt-matthias replied:
I tried to cleanup the PR and removed all the lexicmap stuff

mvdbeek replied:
The command block doesn't look right to me, if the input was flat files or directories it either shouldn't be needed or there is a fix necessary in `CompressedFile`.

mvdbeek replied:
https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/util/compression_utils.py#L257-L259 that seems like it would be a problem, and is rather unpredictable.

bernt-matthias committed:
use specific package version

bernt-matthias committed:
fix converter: allow for flat zip files

zip files whose content is in a single top dir is correctly extracted.
but if the content is contained in the zip's root the extract method will
[add another level (the dataset name)](https://github.com/galaxyproject/galaxy/blob/af2d24082c52cbf8709561044f2082758aec6ff3/lib/galaxy/util/compression_utils.py#L256).
